doc #1 Add README.md

mstniy opened the pull request:
Fixes #1 
Please add statement of contributions

mackeper committed:
feat #17 Add function and test for build branch

mackeper committed:
Merge branch 'master' of github.com:top-tier-ci-coders/ContinuousIntegrationServer into issue/17

mstniy committed:
refactor #6  Remove gradles App.java and its tests

agylling committed:
feat #16 Progress on pull_branch

Pushing my progress for the pull_branch function and the test,
the function still needs to switch branch, then it's done

mstniy committed:
Feat #31 Make handle ignore non-push events

mstniy committed:
doc #1 Add README.md

philippaornell commented on README.md:
```diff
@@ -1,2 +1,44 @@
+## Directory structure
+`src/main/java/CI`: contains the source files of the project.
+
+`src/tes/java/CI`: contains the tests for the project.
```
should be src/test

philippaornell commented on README.md:
```diff
@@ -1,2 +1,44 @@
+ContinuousIntegrationServer is a small continuous integration CI server.
+
+### How does it work?
+The program is written in Java. The project uses the Gradle buils system.
```
Gradle buils -> Gradle Build

philippaornell reviewed (commented):
Needs some spelling errors to be fixed. Otherwise good. Although we will need to add more to this readme at the end of the project

agylling committed:
feat #16 Complete pull_branch function

We fixed the issue with pull_branch function and added proper test code

Worked on by Andreas Gylling and Philippa Örnell

Fixes #16

agylling committed:
Merge branch 'master' into Issue/16

agylling committed:
Merge branch 'master' of https://github.com/top-tier-ci-coders/ContinuousIntegrationServer into Issue/16

agylling committed:
Merge branch 'Issue/16' of https://github.com/top-tier-ci-coders/ContinuousIntegrationServer into Issue/16

philippaornell committed:
Merge pull request #34 from top-tier-ci-coders/Issue/16

Issue/16 The pull_branch function

philippaornell committed:
Merge pull request #25 from top-tier-ci-coders/issue/17

Issue/17

philippaornell committed:
feat #18
The start_test method was implemented and tested. Worked on by Philippa and Andreas (Pair programming)

philippaornell committed:
feat #18 Add start_test method and tests
The start_test method was implemented and tested. Worked on by Philippa and Andreas (Pair programming)

philippaornell committed:
Merge branch 'issue/18' of https://github.com/top-tier-ci-coders/ContinuousIntegrationServer into issue/18

mstniy committed:
Feat #37 Change start_test test case to use "buildstestspass" #37

mstniy committed:
Merge pull request #36 from top-tier-ci-coders/issue/18

Issue/18

mstniy committed:
Fix #39 Avoid using the bash tilda in pull_branch
Also change its test case accordingly

mstniy committed:
Fix #41 Fix pull_branch so that it returns null for nonexistent branches.
Add a test to testPullBranch that tests that
Change testPullBranch so that it uses the dummy branch "buildstestspass", not "test"
In testPullBranch, add code that checks the content of the file "well_known" to make sure that the pull was completed properly.

mstniy committed:
Feat #29 Enable STDOUT in build.gradle for tests

philippaornell committed:
Merge pull request #43 from top-tier-ci-coders/issue/29

Feat #29 Enable STDOUT in build.gradle for tests

philippaornell committed:
Merge pull request #40 from top-tier-ci-coders/issue/39

Fix #39 Avoid using the bash tilda in pull_branch

philippaornell committed:
Merge pull request #42 from top-tier-ci-coders/issue/41

Issue/41

philippaornell committed:
doc #46 Add documentation comment

mstniy committed:
Feat #31 Add javadoc to handlePushEvent and handle functions.

mstniy committed:
Feat #31 Fix typo

mstniy committed:
Fix #48 Re-enable send_notification tests
The test that tests the happy path is still disabled, because the code doesn't seem to be able to send emails outside KTH (maybe some form of authentication is required?).

philippaornell committed:
Merge pull request #47 from top-tier-ci-coders/issue/46

doc #46 Add documentation comment

mackeper committed:
Merge pull request #27 from top-tier-ci-coders/issue/6

refactor #6  Remove gradles App.java and its tests

agylling committed:
Merge pull request #32 from top-tier-ci-coders/issue/31

Feat #31 Make handle ignore non-push events

agylling committed:
feat #36 Add request_push function

Added the core to request_push function and a test case that should be succesful, having troubles with kth SMTP server. Will enter more test cases when I know what to do in order to continue

agylling committed:
feat #35 Add test cases

Added two more test cases where it fails on building for one and testing on the other

Fixes #35

agylling committed:
feat #35 Correct spelling

Changed name of buildSucess to buildSuccess

mstniy committed:
Feat #48 Fix send_notification to use an existing Gmail account

The send_notification success test is still commented out to keep my email from overflowing with spam. Enable it on your working directory if you want to test it.

mackeper committed:
Merge pull request #50 from top-tier-ci-coders/issue/48

Fix #48 Re-enable send_notification tests

mackeper committed:
doc #12 fix the formatting of the title

philippaornell committed:
Merge pull request #53 from top-tier-ci-coders/issue/12

doc #12 fix the formatting of the title

mackeper committed:
feat #55 Add execute permission to gradlew

mstniy committed:
Merge pull request #56 from top-tier-ci-coders/issue/55

feat #55 Add execute permission to gradlew

agylling committed:
feat #35 Improved function

After comments on pull request I've made some changes improving the code

Fixes #35

agylling committed:
Merge branch 'master' of https://github.com/top-tier-ci-coders/ContinuousIntegrationServer into Issue/35

agylling committed:
feat #35 removed redundant condition

Fixes #35

agylling committed:
feat #35 Fixed potential Bug

Fixed a potential bug in Pull_branch that would affect the request_push function

Fixes #35

agylling committed:
Merge pull request #51 from top-tier-ci-coders/Issue/35

Issue/35 Pull request

mstniy committed:
Feat #58 Make request_push more verbose

mstniy committed:
Fix #59 Remove the test involving the master branch from testRequestPush

mstniy committed:
Fix #17 Expand the test case for build_branch function

agylling committed:
Merge pull request #62 from top-tier-ci-coders/issue/59

Fix #59 Remove the test involving the master branch from testRequestPush

mstniy committed:
Merge branch 'master' of github.com:top-tier-ci-coders/ContinuousIntegrationServer into issue/58

mstniy committed:
Merge branch 'master' of github.com:top-tier-ci-coders/ContinuousIntegrationServer into issue/17

agylling committed:
Merge pull request #61 from top-tier-ci-coders/issue/58

Feat #58 Make request_push more verbose

agylling committed:
Merge pull request #63 from top-tier-ci-coders/issue/17

Fix #17 Expand the test case for build_branch function

philippaornell committed:
feat #66 Add log message
Add log message for user, where log can be found

agylling committed:
Merge pull request #68 from top-tier-ci-coders/issue/66

feat #66 Add log message

mstniy committed:
Feat #54 Make request_push return a PipelineResult

It enables more specific testing of request_push

philippaornell committed:
Merge pull request #69 from top-tier-ci-coders/issue/54

Feat #54 Make request_push return a PipelineResult

philippaornell committed:
Merge branch 'issue/1' of https://github.com/top-tier-ci-coders/ContinuousIntegrationServer into issue/1

philippaornell committed:
doc #1 Add documentation and contributions
Add some documentation for how things are implemented and tested. Still some things to fill in.
